### PR TITLE
✨ Source-Greenhouse: Update incremental cursor fields and add custom fields and keyed_custom_fields to Application stream #38236

### PR DESCRIPTION
**Pull request by jimmc2**
## What
Fixes issue described in [this issue](https://github.com/airbytehq/airbyte/issues/38236). Incremental syncs for application stream only return new applications. Fix will return new AND updated applications.

This change also adds missing `custom_fields` and `keyed_custom_fields` to the application stream similar to [this PR](https://github.com/airbytehq/airbyte/pull/30648) for the candidate stream.

## How
The manifest.yaml file was edited to update the incremental sync cursors and the custom field keys were added in the same manner they were for the candidate stream

## User Impact
Positive
- Users will now receive all updates to applications while using incremental syncs.
- Lower number of requests for users currently forced to use full fresh syncs
- Users will now have access to custom fields set on their applications

Negative
- Potential breaking change for users not expecting changes to Applications once received, however it more accurately represents the data that is in the Greenhouse Application
- Higher number of requests potentially for existing incremental users as more applications will fall under the scope of the cursor
- 

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by marcosmarxm — APPROVED** [review]
LGTM need approval from the connector team as the Greenhouse is a certified connector.

**Comment by jimmc2** [comment]
Awesome @marcosmarxm! Thanks so much for looking at this so quickly. Let me know if there's anything else you need from me.

**Comment by marcosmarxm** [comment]
@jimmc2 at the moment nothing more from your side!

File: docs/integrations/sources/greenhouse.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ The Greenhouse connector should not run into Greenhouse API limitations under no
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                                              |
 | :------ | :--------- | :------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 0.5.4   | 2024-05-15 | [38258](https://github.com/airbytehq/airbyte/pull/38258) | Update Application incremental stream cursor and add `custom_field` and `keyed_custom_field` columns                                                                                                                                             |
```
Comment (jimmc2) [inline docs/integrations/sources/greenhouse.md:74]: I saw the other versions were bumped to 1.0 but I still had this one at 0.5.4. Not sure if I should bump this to 1.0 as well.

**Comment by jimmc2** [comment]
Hey @marcosmarxm, I appreciate you jumping on this so quickly. Is there anyone I can reach out to on the Connector team that can help me take it to the next step. 

It appears there is only one failing test and it appears to be because of an internal error unrelated to the PR. 

Thanks again, Jimmy

**Comment by jimmc2** [comment]
Hey @marcosmarxm just wanted to follow up again to see if there is anything else I can do to get this merged. Anyone on the python connectors team I can reach out to help get the approval required to run the last set of failing tests?

**Comment by marcosmarxm** [comment]
@jimmc2 I'll review during the week, sorry the delay here.

**Comment by etiennecallies** [comment]
Hey ! @marcosmarxm me again. It's too bad this PR is not merged, it's perfect!

**Comment by marcosmarxm** [comment]
Hello @etiennecallies I'm going to be on PTO this week but we can work together next week to get this merged.

**Comment by etiennecallies** [comment]
Hi @marcosmarxm, I'm still available to work on this. Contact me here or on my discord `etiennecoh`.

**Comment by jimmc2** [comment]
https://github.com/airbytehq/airbyte/pull/46566 I just duplicated this PR into a new one from the same branch here in case that would help move this along

**Comment by jimmc2** [comment]
@marcosmarxm @etiennecallies
any chance we can take a look at the updated PR this week? thanks!

**Review by etiennecallies — COMMENTED** [review]
Approved ✅

**Pull request closed without merging**